src/jit/gentree.cpp

```diff
@@ -764,6 +764,59 @@ bool GenTree::gtHasReg() const
     return hasReg;
 }
 
+//-----------------------------------------------------------------------------
+// GetRegisterDstCount: Get the number of registers defined by the node.
+//
+// Arguments:
+//    None
+//
+// Return Value:
+//    Reg Mask of GenTree node.
```

Comment:
Is Return Value description correct here?
Do we call this method before the actual register allocation?
What does it return for a tree, that saves its result on stack (RegOptional)?

Reply:
> Is Return Value description correct here?

No, I'm not sure what I was thinking when I wrote that (probably copied from somewhere else). It should be "The number of registers defined by this instruction."

> Do we call this method before the actual register allocation?

This is called after the `TreeNodeInfoInit`, and before actual register allocation to validate the consume/produce numbers. It should probably be under `#ifdef DEBUG`.

> What does it return for a tree, that saves its result on stack (RegOptional)?

It returns the number of registers that it would define assuming that it gets a register. Note that, currently, regOptional is only supported for uses, though issues #7752 and #7753 would include setting it for defs as well.